src/main/java/com/example/demo/exception/ApiException.java

```diff
@@ -0,0 +1,17 @@
+package com.example.demo.exception;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+@Getter
+
+public class ApiException extends RuntimeException {
```

Comment:
Nazwalbym bardziej szczegolowo medical exception mozesz rowniez zastosowac sie do:
Zaprojektowalbym to troche inaczej zrobilbym go jako MedicalException <- i uzywal jako podstawowego wyjatku dla naszego systemu. Co to oznacza?

    taka klasa moglaby byc abstrakcyjna (abstract). -> wtedy poniekad zmusilibysmy ludzi do tworzenia wyjatkow i uzywania bardziej szczegolowych z nich